File: other/remove_adjacent_dup_str.cpp
#include "leetcode.h"

class Solution {
public:
  string removeDuplicates(string s) {
    string ans;
    for (char &c : s) {
      if (ans.size() && c == ans.back())
        ans.pop_back();
      else
        ans.push_back(c);
    }
    return ans;
  }
};

int main() {
  Solution obj;
  cout << obj.removeDuplicates("abbaca");
}

File: other/remove_adjacent_dup_str.rs
struct Solution;

impl Solution {
    pub fn remove_duplicates(s: String) -> String {
        let mut ans = String::new();
        for c in s.chars() {
            if ans.len() > 0 && ans.chars().last() == Some(c) {
                ans.pop();
            } else {
                ans.push(c);
            }
        }
        ans
    }
}

fn main() {
    let s = "abbaca".to_string();
    print!("{}", Solution::remove_duplicates(s));
}
